Mathlib/Data/Sym/Sym2.lean

```diff
@@ -590,6 +590,55 @@ theorem fromRel_toRel (s : Set (Sym2 α)) : fromRel (toRel_symmetric s) = s :=
 
 end Relations
 
+section ToMultiset
+
+/-- Map an unordered pair to an unordered list. -/
+def toMultiset {α : Type*} (z : Sym2 α) : Multiset α := by
+  refine Sym2.lift ?_ z
+  use (Multiset.ofList [·, ·])
+  simp [List.Perm.swap]
+
+/-- Mapping an unordered pair to an unordered list produces a multiset of size `2`. -/
+lemma card_toMultiset {α : Type*} (z : Sym2 α) : z.toMultiset.card = 2 := by
+  induction z
+  simp [Sym2.toMultiset]
+
+/-- The members of an unordered pair are members of the corresponding unordered list. -/
+theorem mem_toMultiset {α : Type*} {x : α} {z : Sym2 α} :
+    x ∈ z ↔ x ∈ (z.toMultiset : Multiset α) := by
+  induction z
+  simp [Sym2.toMultiset]
+
+end ToMultiset
+
+section ToFinset
+
+variable [DecidableEq α]
+
+/-- Map an unordered pair to a finite set. -/
+def toFinset (z : Sym2 α) : Finset α := (z.toMultiset : Multiset α).toFinset
+
+/-- The members of an unordered pair are members of the corresponding finite set. -/
+theorem mem_toFinset {x : α} {z : Sym2 α} : x ∈ z ↔ x ∈ z.toFinset := by
+  rw [Sym2.mem_toMultiset, Sym2.toFinset, Multiset.mem_toFinset]
+
+lemma toFinset_mk_eq {x y : α} : s(x, y).toFinset = {x, y} := by
+  ext; simp [←Sym2.mem_toFinset, ←Sym2.mem_iff, Sym2.mem_toMultiset]
+
+/-- Mapping an unordered pair on the diagonal to a finite set produces a finset of size `1`. -/
+theorem card_toFinset_of_isDiag (z : Sym2 α) (h : z.IsDiag) : #(z : Sym2 α).toFinset = 1 := by
```

Comment:
Let's have a version of these lemmas of the form `_ = if z.IsDiag then 1 else 2`